std/utf.d

```diff
@@ -3802,17 +3802,12 @@ template byUTF(C) if (isSomeChar!C)
         {
             static struct Result
             {
-                this(ref R r)
-                {
-                    this.r = r;
-                }
-
                 @property bool empty()
                 {
                     return pos == fill && r.empty;
                 }
 
-                @property auto front()
+                @property auto front() scope // 'scope' required by call to decodeFront() below
```

Comment:
So I'm trying to read and understand the DIP, please correct me on any details I have wrong:

Adding scope here adds checks to the `this` reference to the struct itself to make sure references to member variables aren't escaped from the function calls and return values. This does not change the return value of the function and this change should be completely transparent to current code.